pkg/clusteragent/admission/mutate/autoinstrumentation/mutators.go

```diff
@@ -17,21 +17,31 @@ import (
 
 // containerMutator describes something that can mutate a container.
 type containerMutator interface {
+	podMutator
+
```

Comment:
I don't think this is the right abstraction. It would requires every `containerMutator` to reimplement pod mutation, which requires duplicating this everywhere:
```
func (...) mutatePod(pod *corev1.Pod) error {
	return mutatePodContainers(pod, r)
}
```
and all it gets is turning this:
```
mutatePodContainers(pod, r)
```
into this:
```
r.mutatePod(pod)
```

It's fine to just have `mutatePodContainers(pod, r)` take care of executing a `containerMutator` across a pod. If we absolutely needed it, we could add a `podContainerMutator(c containerMutator) podMutator` function that would wrap it, but I don't know that it's necessary.

Reply:
Removed this.